Softeq.XToolkit.WhiteLabel.Droid/ImagePicker/ImagePickerActivity.cs

```diff
@@ -0,0 +1,301 @@
+            {
+                ICursor cursor = null;
+                try
+                {
+                    cursor = context.ContentResolver.Query(uri, new[] { MediaStore.MediaColumns.Data }, null, null, null);
+                    if (cursor == null || !cursor.MoveToNext())
+                    {
+                        return null;
+                    }
+
+                    var column = cursor.GetColumnIndex(MediaStore.MediaColumns.Data);
+                    string contentPath = null;
+
+                    if (column != -1)
+                    {
+                        contentPath = cursor.GetString(column);
+                    }
+
+                    if (contentPath == null || !contentPath.StartsWith(FileScheme, StringComparison.InvariantCultureIgnoreCase))
+                    {
+                        string fileName = null;
+                        try
+                        {
+                            fileName = System.IO.Path.GetFileName(contentPath);
+                        }
+                        catch (Exception ex)
+                        {
+                            System.Diagnostics.Debug.WriteLine("Unable to get file path name, using new unique " + ex);
+                        }
+
+                        contentPath = CopyContentToFile(context, uri, fileName);
+                    }
+
+                    return contentPath;
+                }
+                finally
+                {
+                    cursor?.Close();
+                    cursor?.Dispose();
+                }
+            }, CancellationToken.None, TaskCreationOptions.None, TaskScheduler.Default);
+        }
+
+        private string CopyContentToFile(Context context, Android.Net.Uri uri, string fileName)
+        {
+            string contentPath = null;
+            var outputPath = GetOutputMediaFile(context, TempFolder, fileName);
+
+            try
+            {
+                using (var input = context.ContentResolver.OpenInputStream(uri))
+                {
+                    using (var output = File.Create(outputPath.Path))
+                    {
+                        input.CopyTo(output);
+                    }
+                }
+
+                contentPath = outputPath.Path;
+            }
+            catch (Java.IO.FileNotFoundException fnfEx)
+            {
+                System.Diagnostics.Debug.WriteLine("Unable to save picked file from disk " + fnfEx);
+            }
+
+            return contentPath;
+        }
+
+        private Android.Net.Uri GetOutputMediaFile(Context context, string subdir, string name)
+        {
+            subdir = subdir ?? string.Empty;
+
+            if (string.IsNullOrWhiteSpace(name))
+            {
+                var timestamp = DateTime.Now.ToString("yyyyMMdd_HHmmss", CultureInfo.InvariantCulture);
+                name = "IMG_" + timestamp + ".jpg";
+            }
+
+            var mediaType = Android.OS.Environment.DirectoryPictures;
```

Comment:
need to discuss where we save an image:
- in global user pictures
- in external for app (when removing)

Reply:
After discussed with @sleushunou we will use cache